vllm/v1/core/sched/scheduler.py

```diff
@@ -824,7 +825,8 @@ def make_spec_decoding_stats(
         if not self.log_stats:
             return None
         if spec_decoding_stats is None:
-            spec_decoding_stats = SpecDecodingStats()
-        spec_decoding_stats.observe(num_draft_tokens=num_draft_tokens,
-                                    num_accepted_tokens=num_accepted_tokens)
+            spec_decoding_stats = SpecDecodingStats.new(self.num_spec_tokens)
```

Comment:
Here we have a new requirement for `num_speculative_tokens` to be set in the config whenever we're recording spec decoding metrics